### PR TITLE
fix(gtk): correct `drag_window` signature, closes #224

### LINKED ISSUE #224: `drag_window` API is not consistent
On Linux, it returns `()`: https://github.com/tauri-apps/wry/blob/cea79fb7098cf870274b24512b117bd220ca7f34/src/application/window.rs#L717
But the winit interface returns a Result: https://docs.rs/tauri-winit/0.24.1/winit/window/struct.Window.html#method.drag_window

### PR DESCRIPTION
**Pull request by amrbashir**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes. Issue #___
- [ ] No

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

File: src/application/window.rs [hunk 1]
```diff
@@ -714,13 +714,14 @@ impl Window {
     self.maximized.load(Ordering::Acquire)
   }
 
-  pub fn drag_window(&self) {
+  pub fn drag_window(&self) -> Result<(), ExternalError> {
     if let Err(e) = self
```
Comment (lucasfernog) [inline src/application/window.rs:718]: should we propagate this error instead?

Comment (amrbashir) [reply src/application/window.rs:718]: why ? `ExternalError` indicates its either a OS error, or just not a supported method and the error here is `SendError`.

Comment (lucasfernog) [reply src/application/window.rs:718]: yeah, and also i think the SendError should never happen right

Comment (wusyong) [reply src/application/window.rs:718]: SendError occurs only when the receiver is disconnected first. This depends on how gtk handles its threads and tasks.

**Review by wusyong — APPROVED** [review]

**Pull request merged**